File: go/accounts.go
package main

import (
	"fmt"
	"os"

	dd "github.com/deltadefi-protocol/go-sdk"
	"github.com/lpernett/godotenv"
)

func main() {
	accounts()
}

func accounts() {
	godotenv.Load()
	config := dd.ApiConfig{
		Network:           "staging",
		ApiKey:            os.Getenv("DELTADEFI_API_KEY"),
		OperationPasscode: os.Getenv("ENCRYPTION_PASSCODE"),
	}

	client := dd.NewDeltaDeFi(config)

	accountBalanceRes, _ := client.Accounts.GetAccountBalance()
	fmt.Println("\nAccount Balance:")
	for _, record := range *accountBalanceRes {
		fmt.Println(record)
	}

	fmt.Println("\nDeposit Records:")
	depositRecordRes, _ := client.Accounts.GetDepositRecords()
	for _, record := range *depositRecordRes {
		fmt.Println(record)
	}

	fmt.Println("\nWithdrawal Records:")
	withdrawalRecordRes, _ := client.Accounts.GetWithdrawalRecords()
	if len(*withdrawalRecordRes) == 0 {
		fmt.Println("No withdrawal records found")
	}
	for _, record := range *withdrawalRecordRes {
		fmt.Println(record)
	}

	fmt.Println("\nOrder Records:")
	orderRecordsRes, _ := client.Accounts.GetOrderRecords(&dd.GetOrderRecordRequest{Status: dd.OrderRecordStatusOpenOrder})
	for _, record := range orderRecordsRes.Data {
		fmt.Println(record)
	}

	fmt.Println("\nOrder Record for:", orderRecordsRes.Data[0].Orders[0].OrderID)
	orderRecordRes, _ := client.Accounts.GetOrderRecord(orderRecordsRes.Data[0].Orders[0].OrderID)
	fmt.Printf("Order Record: %v\n", orderRecordRes)

}

File: rust/accounts.rs
use deltadefi::{DeltaDeFi, Stage};
use dotenv::dotenv;
use std::env;

pub async fn accounts() {
    dotenv().ok();
    let api_key = env::var("DELTADEFI_API_KEY").unwrap();
    let deltadefi = DeltaDeFi::new(api_key, Stage::Staging, None).unwrap();

    let account_balance = deltadefi.accounts.get_account_balance().await.unwrap();
    println!("\nAccount Balance: {:?}", account_balance);

    let deposit_records = deltadefi.accounts.get_deposit_records().await.unwrap();
    println!("\nDeposit Records: {:?}", deposit_records);

    let withdrawal_records = deltadefi.accounts.get_withdrawal_records().await.unwrap();
    println!("\nWithdrawal Records: {:?}", withdrawal_records);

    let order_records = deltadefi
        .accounts
        .get_order_records(deltadefi::OrderRecordStatus::OpenOrder, None, None, None)
        .await
        .unwrap();
    println!("\nOrder Records: {:?}", order_records);

    let order_record = deltadefi
        .accounts
        .get_order_record(&order_records.data[0].orders[0].order_id)
        .await
        .unwrap();
    println!("\nOrder Record: {:?}", order_record);
}
